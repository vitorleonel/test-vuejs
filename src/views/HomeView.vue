<script lang="ts">
import { defineComponent } from 'vue'

import SButton from '@/components/SButton.vue'
import SInput from '@/components/SInput.vue'
import SSelect from '@/components/SSelect.vue'

export default defineComponent({
  components: { SButton, SInput, SSelect }
})
</script>

<template>
  <SButton text="My Button" variant="default" />
  <SInput />
  <SSelect
    :options="[
      { value: 1, label: 'Option 1' },
      { value: 2, label: 'Option 2' },
      { value: 3, label: 'Option 3' },
      { value: 4, label: 'Option 4' },
      { value: 5, label: 'Option 5' }
    ]"
  />
</template>
