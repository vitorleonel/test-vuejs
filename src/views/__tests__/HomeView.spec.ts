import { mount, type VueWrapper } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import HomeView from '@/views/HomeView.vue'

describe('HomeView', () => {
  it('should render properly', () => {
    const wrapper: VueWrapper = mount(HomeView)

    expect(wrapper.element).toMatchSnapshot()
  })
})
