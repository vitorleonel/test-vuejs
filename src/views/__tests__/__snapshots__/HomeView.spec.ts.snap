// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView > should render properly 1`] = `
<div
  data-v-app=""
>
  
  <header
    class="header"
    data-v-51353816=""
    data-v-81cf18a8=""
  >
    <div
      class="container"
      data-v-51353816=""
    >
      <h1
        data-v-51353816=""
      >
        BestBuy
      </h1>
      <nav
        data-v-51353816=""
      >
        <router-link
          data-v-51353816=""
          to="/"
        >
          Home
        </router-link>
        <router-link
          data-v-51353816=""
          to="/shop"
        >
          Shop
        </router-link>
        <router-link
          data-v-51353816=""
          to="/about"
        >
          About us
        </router-link>
      </nav>
    </div>
  </header>
  <main
    data-v-81cf18a8=""
  >
    
    <section
      class="home container"
      data-v-b4e148ca=""
    >
      <img
        alt="Everything you need"
        class="home__banner"
        data-v-b4e148ca=""
        loading="lazy"
        src="/src/assets/home-banner.png"
      />
      <nav
        class="home__categories"
        data-v-b4e148ca=""
      >
        
        <routerlink
          class="home__category"
          data-v-b4e148ca=""
          to="/shop"
        >
          <h3
            data-v-b4e148ca=""
          >
            Clothes
          </h3>
          <span
            data-v-b4e148ca=""
          >
            The best clothes you can find on the internet
          </span>
        </routerlink>
        <routerlink
          class="home__category"
          data-v-b4e148ca=""
          to="/shop"
        >
          <h3
            data-v-b4e148ca=""
          >
            Electronics
          </h3>
          <span
            data-v-b4e148ca=""
          >
            Exclusive prices on all kind of electronics
          </span>
        </routerlink>
        <routerlink
          class="home__category"
          data-v-b4e148ca=""
          to="/shop"
        >
          <h3
            data-v-b4e148ca=""
          >
            Automotive
          </h3>
          <span
            data-v-b4e148ca=""
          >
            Everything you need for your car at a great price
          </span>
        </routerlink>
        <routerlink
          class="home__category"
          data-v-b4e148ca=""
          to="/shop"
        >
          <h3
            data-v-b4e148ca=""
          >
            Personal care
          </h3>
          <span
            data-v-b4e148ca=""
          >
            Take care of yourself and on people you love
          </span>
        </routerlink>
        
      </nav>
    </section>
    
  </main>
  
</div>
`;
